File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        count =0
        curr = head
        while curr :
            count+=1
            curr=curr.next

        if n==count:
            return head.next
        k= count-n
        i=0
        curr=head
        while i < k-1:
            curr=curr.next
            i+=1

        toDel = curr.next
        curr.next=toDel.next
        del toDel
        return head

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
         vector<int> outList;
    for (auto head : lists)
    {
        ListNode * curr =head;
        while (curr)
        {
            outList.push_back(curr->val);
            curr=curr->next;
        }
    }

    sort(outList.begin(),outList.end());

    ListNode * dummy=new ListNode();
    ListNode * curr=dummy;

    for (int i = 0; i < outList.size(); ++i)
    {
        ListNode * node = new ListNode(outList[i]);
        curr->next=node;
        curr=curr->next;

    }
    return dummy->next;
    }
};
